### PR TITLE
Decorator takes an input and arguments it with extensions.

### PR DESCRIPTION
**Pull request by n3wscott**
This is a POC.

### CODE REVIEW

File: config/300-decorator.yaml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# Copyright 2019 The Knative Authors
```
Comment (nachocano) [inline config/300-decorator.yaml]: Copyright 2019 Google LLC

File: pkg/apis/messaging/v1alpha1/decorator_types.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/apis"
+	duckv1beta1 "knative.dev/pkg/apis/duck/v1beta1"
+	"knative.dev/pkg/webhook"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Decorator TODO
```
Comment (nachocano) [inline pkg/apis/messaging/v1alpha1/decorator_types.go]: can you add the documentation in this PR?

File: pkg/apis/messaging/v1alpha1/decorator_types.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+type DecoratorSpec struct {
+	// Extensions will specify what is added to an event as an extension.
+	// Each key-value pair is added independently.
+	Extensions map[string]string `json:"extensions,omitempty"`
+	// Sink is a reference to an object that will resolve to a domain name to use
+	// as the sink.
+	Sink *corev1.ObjectReference `json:"sink,omitempty"`
+}
+
+// Save for later:
```
Comment (nachocano) [inline pkg/apis/messaging/v1alpha1/decorator_types.go]: outdated? can remove?

File: config/300-decorator.yaml (hunk 1 above)
Comment (n3wscott) [reply config/300-decorator.yaml]: I have been transitioning back to Knative Authors

**Comment by nachocano** [comment]
Ohh just saw it was a POC. Discard all the cosmetic comments

File: cmd/decorator/main.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
+/*
```
Comment (vaikas-google) [inline cmd/decorator/main.go]: what's the copyright insoncistency about? seems like we should be consistent :)

Comment (n3wscott) [reply cmd/decorator/main.go]: fixed.

**Pull request merged**